### DISCUSSION TITLE
webpack multiple entry points array order by

### DISCUSSION THREAD

**Discussion by kp1007** [webpack/webpack#14162] [Q&A]
![image](https://user-images.githubusercontent.com/10395927/132319248-54b1aa2c-880e-43c4-962d-9b4b42a04de5.png)

Hi,
here I have bunch of files which i want to create sing bundle , I wish to use three.js as a first 
but after bundle created it not ordered in sequence which I provided in array hence i am getting error because threejs must be top it is dependable file of others 
please suggest for solution  ?

**Comment by alexander-akait** [comment]
Please provide steps to reproduce

**Reply by kp1007** [reply]
just create multiple entrypoint array:

u can download these file and use them in to array

here i provided two files 
1)three.js 2)detector.js
https://cdnjs.cloudflare.com/ajax/libs/three.js/r128/three.js
https://cdn.jsdelivr.net/npm/detector-js@1.3.2/dist/detector.js
![image](https://user-images.githubusercontent.com/10395927/132340466-15d587ad-04ad-424f-9148-002ce62a9267.png)

and then I run them to create bundle and it created, both files are merged to one but detector.js is first, but this script is totally dependent with three.js so, it should me merge with proper ordering.

![image](https://user-images.githubusercontent.com/10395927/132340567-adbba451-8fda-4076-9a4b-13dc6fb8f0f8.png)

**Reply by alexander-akait** [reply]
It is merged with proper order

**Reply by alexander-akait** [reply]
You don't provide version of webpack, and steps to reproduce, hard to say what is wrong

**Reply by kp1007** [reply]
i added snap where u can check script is started from line no: 12 and these come from detector.js, so, instead of it i want three.js code first

**Reply by kp1007** [reply]
"webpack": "^5.51.1",

**Reply by alexander-akait** [reply]
Can't reproduce, order is right

**Reply by kp1007** [reply]
ok will upload zip file soon :-)

**Reply by kp1007** [reply]
[
[webpack-static.zip](https://github.com/webpack/webpack/files/7121418/webpack-static.zip)
](url)

**Reply by kp1007** [reply]
during merge it ordering bundle by alphabetical file names and it also take first preference of capital file name

**Reply by alexander-akait** [reply]
As you can see using `devtool: false`, order is valid:
```
/******/ 	__webpack_require__("./src/three.js");
/******/ 	var __webpack_exports__ = __webpack_require__("./src/Detector.js");
```

**Reply by kp1007** [reply]
hard to explain/convince  you, it throwing error on that bundle when i use this bundle on page
![image](https://user-images.githubusercontent.com/10395927/132384477-7a956f1f-8f0a-47b3-8035-ed39af3e9f0d.png)

**Reply by alexander-akait** [reply]
What is stack trace?

**Reply by kp1007** [reply]
my issue  is similar like    https://github.com/webpack/webpack/issues/8264   and i can see that was also unresolved and closed ;-9, one js is depended to other hence it require to load one js first.  as per sequence are set

**Reply by alexander-akait** [reply]
Please create reproducible test repo and provide steps to reproduce the problem and I will help you

**Reply by kp1007** [reply]
[webpack-staticthree.zip](https://github.com/webpack/webpack/files/7126370/webpack-staticthree.zip)
run photos.html file and check console you will get idea of error

**Reply by alexander-akait** [reply]
Because you don't have `THREE` as global variable, `three.js` uses wrapper:
```
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports) :
        typeof define === 'function' && define.amd ? define(['exports'], factory) :
        (factory((global.THREE = global.THREE || {})));
}(this, (function (exports) {
 // Code
});
```

In your case `typeof exports === 'object' && typeof module !== 'undefined'` is `true`, there are many solutions:
1. 
```
entry: {
    // jebsite: ["./src/jquery.min.js"],
     testthree:["./src/OBJLoader.js"],
    //website: './src/join.js'
  },
```

And

```
import * as THREE from "./three";

window.THREE = THREE;
window.THREE.OBJLoader = ( function () {
```
2. Using `external` and load this from CDN
3. Use `three.js` from npm and use ECMA modules version
4. Change `three.js` wrapper
